colearn_pytorch/pytorch_learner.py

```diff
@@ -159,7 +198,16 @@ def mli_propose_weights(self) -> Weights:
         self.train()
         new_weights = self.mli_get_current_weights()
         self.set_weights(current_weights)
-        # new_weights.training_summary = ...
+
```

Comment:
I think the best and fastest behaviour would be the following:
Before the training step we have old_epsilon (spent), old_weights
1.) check what would be the new_epsilon if we were running an additional train (this should run quickly)
2.) if new_epsilon > target_epsilon: return old_weights & old_epsilon & error_code:DP_BUDGET_EXCEEDED 
     else: train & return new_weights & new epsilon

Since we only used old_epsilon amount of privacy, and the error code indicates, that training further would result in budget exceeding.

Or if it is difficult to do it in opacus, then:
Before the training step we have old_epsilon (spent), old_weights
0.) train
1.) calculate the new_epsilon
2.) if new_epsilon > target_epsilon: return old_weights & old_epsilon & error_code:DP_BUDGET_EXCEEDED 
     else: return new_weights & new epsilon

Also does the job, but a bit slower.